internal/webhooks/cluster.go

```diff
@@ -268,18 +275,11 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 	}
 
 	// Get the ClusterClass referenced in the Cluster.
-	clusterClass, clusterClassPollErr := webhook.pollClusterClassForCluster(ctx, newCluster)
-	if clusterClassPollErr != nil &&
-		// If the error is anything other than "NotFound" or "NotReconciled" return all errors at this point.
-		!(apierrors.IsNotFound(clusterClassPollErr) || errors.Is(clusterClassPollErr, errClusterClassNotReconciled)) {
-		allErrs = append(
-			allErrs, field.InternalError(
-				fldPath.Child("class"),
-				clusterClassPollErr))
-		return allErrs
-	}
+	clusterClass, warnings, clusterClassPollErr := webhook.validateClusterClassExistsAndIsReconciled(ctx, newCluster)
+	allWarnings = append(allWarnings, warnings...)
+
```

Comment:
Not sure if it's safe to drop this. What if we couldn't get the ClusterClass with another error than not found (e.g. connection errors or whatever else might happen). It seems to me that ClusterClassesAreCompatible will panic then

Reply:
You're right - I've added tests to cover this case and I've re-added the code which returns here if there's an unrecognized error.

Reply:
Thx!